lib/s3_helper.rb

```diff
@@ -264,7 +264,12 @@ def default_s3_options
   end
 
   def get_path_for_s3_upload(path)
-    path = File.join(@s3_bucket_folder_path, path) if @s3_bucket_folder_path && path !~ /^#{@s3_bucket_folder_path}\//
+    if @s3_bucket_folder_path &&
+        !path.starts_with?(@s3_bucket_folder_path) &&
```

Comment:
I think starts_with is a lot more readable here than regex and does the same thing, existing tests pass fine.